controllers/infinispan_controller.go

```diff
@@ -302,20 +302,22 @@ func (reconciler *InfinispanReconciler) Reconcile(ctx context.Context, ctrlReque
 	// If an overlay configuration is specified wait for the related ConfigMap
 	overlayConfigMap := &corev1.ConfigMap{}
 	var overlayConfigMapKey string
+	var overlayLog4jConfig bool
 	if infinispan.Spec.ConfigMapName != "" {
 		if result, err := kube.LookupResource(infinispan.Spec.ConfigMapName, infinispan.Namespace, overlayConfigMap, infinispan, r.Client, reqLogger, r.eventRec, r.ctx); result != nil {
 			return *result, err
 		}
-		var foundKey bool
-		// Loop through the data looking for something like xml,json or yaml
+		// Loop through the data looking for something like xml, json or yaml
 		for overlayConfigMapKey = range overlayConfigMap.Data {
 			if overlayConfigMapKey == "infinispan-config.xml" || overlayConfigMapKey == "infinispan-config.json" || overlayConfigMapKey == "infinispan-config.yaml" {
-				foundKey = true
 				break
 			}
 		}
-		if !foundKey {
-			err := fmt.Errorf("infinispan-copnfig.[xml|yaml|json] configuration not found in ConfigMap: %s", overlayConfigMap.Name)
+		// Check if the user added a custom log4j.xml config
+		_, overlayLog4jConfig = overlayConfigMap.Data["log4j.xml"]
+
+		if overlayConfigMapKey == "" && !overlayLog4jConfig {
+			err := fmt.Errorf("infinispan-config.[xml|yaml|json] or log4j.xml configuration not found in ConfigMap: %s", overlayConfigMap.Name)
```

Comment:
@ryanemerson , @leo8a please double check this snip. Still sounds weird to me. Maybe I'm reading wrong, but:
- it supposes that if `infinispan-config` is missing then `overlayConfigMapKey == ""` but it seems to me that it will be equal to the last key iterated in the map;
- plus looks like the error is returned only if both config AND log4j are missing.

Reply:
> plus looks like the error is returned only if both config AND log4j are missing.

This is expected. It should be possible for the user to provide 1 . Custom server config, 2. log4j.xml. 3. Both

> it supposes that if infinispan-config is missing then overlayConfigMapKey == "" but it seems to me that it will be equal to the last key iterated in the map;

Good spot! The easiest thing is to reintroduce the `foundKey` var, or we can update the `for` to use a different var name and then explicitly set `overlayConfigMapKey` if a key is found.

@leo8a can you create a follow up PR?

Reply:
> This is expected. It should be possible for the user to provide 1 . Custom server config, 2. log4j.xml. 3. Both

oh ok, I would fix the `or` in the error message then

Reply:
Yeah maybe we can be more explicit and say something like:

"One of infinispan-config.[xml|yaml|json] or  log4j.xml must be present in the provided ConfigMap"

Reply:
sure, this is being fixed by [PR#1451](https://github.com/infinispan/infinispan-operator/pull/1451), thanks.